File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第3章：空间变换/3.1.8-旋转函数rotate/rotate_OpenCV3.cpp
#include<opencv2/core/core.hpp>
#include<opencv2/highgui/highgui.hpp>
#include<opencv2/imgproc/imgproc.hpp>
using namespace cv;
int main(int argc, char*argv[])
{
	//读入图像
	Mat img = imread(argv[1], IMREAD_ANYCOLOR);
	//旋转 90 180 270 
	Mat rImg;
	rotate(img, rImg, ROTATE_90_CLOCKWISE);
	//显示原图和旋转的结果
	imshow("原图", img);
	imshow("旋转", rImg);
	waitKey(0);
	return 0;
}

File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第3章：空间变换/3.1.8-旋转函数rotate/rotate_OpenCV3.py
# -*- coding: utf-8 -*-
import cv2
import sys
#主函数
if __name__ == "__main__":
    if len(sys.argv)>1:
        image = cv2.imread(sys.argv[1],cv2.IMREAD_ANYCOLOR)
    else:
        print "Usage: python rotate.py image"
    #显示原图
    cv2.imshow("image",image)
    #图像旋转 ： cv2.ROTATE_180  cv2.ROTATE_90_COUNTERCLOCKWISE
    rImg = cv2.rotate(image,cv2.ROTATE_90_COUNTERCLOCKWISE)
    #显示旋转的结果
    cv2.imshow("rImg",rImg)
    cv2.waitKey(0)
    cv2.destroyAllWindows()
    
     